core/src/main/scala/spinal/core/Formal.scala

```diff
@@ -7,18 +7,19 @@ import spinal.core.internals.Operator
 object Formal {
 
   //that.formalPast(delay) replacement
-  def past[T <: Data](that : T, delay : Int)(moreAction: (Int, T) => Unit) : T = {
+  def past[T <: Data](that : T, delay : Int, action: (Int, T) => Unit = null) : T = {
```

Comment:
I'm not realy for it, see the bottom of https://github.com/SpinalHDL/SpinalHDL/discussions/689#discussioncomment-2628742

Reply:
i mean, having action to past.

Reply:
I just revert them.